ImperatorToCK3/Imperator/Countries/CountryFactory.cs

```diff
@@ -11,122 +11,145 @@ public partial class Country {
 				}
 			});
-			parser.RegisterKeyword("family", reader =>
-				country.parsedFamilyIds.Add(reader.GetULong())
-			);
-			parser.RegisterKeyword("minor_family", reader =>
-				country.parsedFamilyIds.Add(reader.GetULong())
-			);
-			parser.RegisterKeyword("monarch", reader =>
-				country.monarchId = reader.GetULong()
-			);
-			parser.RegisterKeyword("ruler_term", reader =>
-				country.RulerTerms.Add(RulerTerm.Parse(reader))
-			);
-			parser.RegisterRegex(monarchyLawRegexStr, reader =>
-				country.monarchyLaws.Add(reader.GetString())
-			);
-			parser.RegisterRegex(republicLawRegexStr, reader =>
-				country.republicLaws.Add(reader.GetString())
-			);
-			parser.RegisterRegex(tribalLawRegexStr, reader =>
-				country.tribalLaws.Add(reader.GetString())
-			);
+			parser.RegisterKeyword("family", reader => parsedCountry.parsedFamilyIds.Add(reader.GetULong()));
+			parser.RegisterKeyword("minor_family", reader => parsedCountry.parsedFamilyIds.Add(reader.GetULong()));
+			parser.RegisterKeyword("monarch", reader => parsedCountry.monarchId = reader.GetULong());
+			parser.RegisterKeyword("ruler_term", reader => parsedCountry.RulerTerms.Add(RulerTerm.Parse(reader)));
+			parser.RegisterRegex(monarchyLawRegexStr, reader => parsedCountry.monarchyLaws.Add(reader.GetString()));
+			parser.RegisterRegex(republicLawRegexStr, reader => parsedCountry.republicLaws.Add(reader.GetString()));
+			parser.RegisterRegex(tribalLawRegexStr, reader => parsedCountry.tribalLaws.Add(reader.GetString()));
 			parser.RegisterRegex(CommonRegexes.Catchall, (reader, token) => {
 				IgnoredTokens.Add(token);
 				ParserHelpers.IgnoreItem(reader);
 			});
 		}
 		public static Country Parse(BufferedReader reader, ulong countryId) {
-			country = new Country(countryId);
+			parsedCountry = new Country(countryId);
 			parser.ParseStream(reader);
-			return country;
+			return parsedCountry;
+		}
+
+		public static void LoadGovernments(Configuration config, List<Mod> mods) {
+			string governmentName = string.Empty;
+			bool typeSpecified = false;
+
+			var governmentParser = new Parser();
+			governmentParser.RegisterKeyword("type", reader => {
+				switch (reader.GetString()) {
+					case "republic":
+						AddRepublicGovernment(governmentName);
+						typeSpecified = true;
+						break;
+					case "monarchy":
+						AddMonarchyGovernment(governmentName);
+						typeSpecified = true;
+						break;
+					case "tribal":
+						AddTribalGovernment(governmentName);
+						typeSpecified = true;
+						break;
+					default:
+						AddMonarchyGovernment(governmentName);
+						break;
+				}
+			});
+
+			var fileParser = new Parser();
+			fileParser.RegisterRegex(CommonRegexes.String, (reader, govName) => {
+				typeSpecified = false;
+				governmentName = govName;
+				governmentParser.ParseStream(reader);
+				if (!typeSpecified) {
+					// monarchy is the default type
+					AddMonarchyGovernment(governmentName);
+				}
```

Comment:
Isn't this already handled by the default case above? If so, can you remove this and the typeSpecified variable?

Reply:
With the switch only being executed on "type" keyword found, no. But I've rewritten the code a bit and the variable is no longer needed.